Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Themes/InteractableColorTheme.cs

```diff
@@ -58,20 +88,211 @@ public override void SetValue(InteractableThemeProperty property, int index, flo
 
+            // check if a text object exists and set the color,
+            // if not then fall back to renderer based color setting.
+            if (SetColorValue != null)
+            {
+                SetColorValue(color);
+            }
+            else
+            {
+                if (SetTextMeshProColor(color))
+                {
+                    SetColorValue = SetTextMeshProColor;
+                    return;
+                }
+
+                if (SetTextMeshProUGUIColor(color))
+                {
+                    SetColorValue = SetTextMeshProUGUIColor;
+                    return;
+                }
+
+                if (SetTextMeshColor(color))
+                {
+                    SetColorValue = SetTextMeshColor;
+                    return;
+                }
+
+                if (SetTextColor(color))
+                {
+                    SetColorValue = SetTextColor;
+                    return;
+                }
+            }
+
+            base.SetValue(property, index, percentage);
+
+        }
+
+        /// <summary>
+        /// Get color on UI Text
+        /// </summary>
+        /// <param name="success"></param>
+        /// <returns></returns>
+        protected Color GetTextColor(out bool success)
+        {
+            Color colour = Color.white;
+            Text text = Host.GetComponent<Text>();
+            if (text != null)
+            {
+                success = true;
+                return text.color;
+            }
+            success = false;
+            return colour;
+        }
+
+        /// <summary>
+        /// Get color from TextMesh
+        /// </summary>
+        /// <param name="success"></param>
+        /// <returns></returns>
+        protected Color GetTextMeshColor(out bool success)
+        {
+            Color colour = Color.white;
+            TextMesh mesh = Host.GetComponent<TextMesh>();
             if (mesh != null)
             {
-                mesh.color = color;
-                return;
+                success = true;
+                return mesh.color;
+            }
+            success = false;
+            return colour;
+        }
+
+        /// <summary>
+        /// Get color from TextMeshPro
+        /// </summary>
+        /// <param name="success"></param>
+        /// <returns></returns>
+        protected Color GetTextMeshProColor(out bool success)
+        {
+            Color colour = Color.white;
+            TextMeshPro tmp = Host.GetComponent<TextMeshPro>();
+            if (tmp)
+            {
+                success = true;
+                return tmp.color;
+            }
+            success = false;
+            return colour;
+        }
+
+        /// <summary>
+        /// Get color from TextMeshProUGUI
+        /// </summary>
+        /// <param name="success"></param>
+        /// <returns></returns>
+        protected Color GetTextMeshProUGUIColor(out bool success)
```

Comment:
Super-duper nit: I've never seen this pattern - typically you'd return success value and use an out for Color.

Reply:
Good point, not sure why it's reversed.